File: srp_estudiante.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

// Clase 1: Estudiante
class Estudiante {
private:
    string nombre;
    int edad;
public:
    Estudiante(string n, int e) {
        nombre = n;
        edad = e;
    }

    string getNombre() {
        return nombre;
    }

    int getEdad() {
        return edad;
    }
};

// Clase 2: DatosEstudiantes (solo muestra datos)
class DatosEstudiantes {
public:
    void mostrar(Estudiante e) {
        cout << "Nombre: " << e.getNombre() << endl;
        cout << "Edad: " << e.getEdad() << endl;
    }
};

// Clase 3: GuardarDatos (solo guarda datos)
class GuardarDatos {
public:
    void guardar(Estudiante e) {
        ofstream archivo("datos_estudiantes.txt", ios::app);
        if (archivo.is_open()) {
            archivo << e.getNombre() << ", " << e.getEdad() << endl;
            archivo.close();
        } else {
            cout << "No se pudo abrir el archivo." << endl;
        }
    }
};

// Programa principal
int main() {
    Estudiante est("Florencia", 25);

    DatosEstudiantes datos;
    datos.mostrar(est);

    GuardarDatos guardar;
    guardar.guardar(est);

    return 0;
}

File: srp_estudiante.py
# Clase 1: Estudiante - Tiene la responsabilidad de representar un estudiante
class Estudiante:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

# Clase 2: DatosEstudiantes - Tiene la responsabilidad de mostrar los datos
class DatosEstudiantes:
    def mostrar(self, estudiante):
        print(f"Nombre: {estudiante.nombre}")
        print(f"Edad: {estudiante.edad}")

# Clase 3: GuardarDatos - Tiene la responsabilidad de guardar los datos (simulado)
class GuardarDatos:
    def guardar(self, estudiante):
        with open("datos_estudiantes.txt", "a") as archivo:
            archivo.write(f"{estudiante.nombre}, {estudiante.edad}\n")

# --- Uso del código ---
# Creamos un estudiante
est1 = Estudiante("Florencia", 25)

# Mostramos sus datos
datos = DatosEstudiantes()
datos.mostrar(est1)

# Guardamos los datos
guardador = GuardarDatos()
guardador.guardar(est1)
